### ISSUE TITLE
Raouls <important> dialogue doesn't play/is shown but shouldn't play.

### ISSUE THREAD

**Issue by degenerated1123** [REGoth-project/REGoth#128]
He also starts with "What do you want this time?!", so probably the wrong <important> is picked.

**Comment by markusobi** [comment]
It seems like the game expects the NPC to not be in ZS_TALK state yet when AI_ProcessInfos is called. `<important>` dialogs, that get not played when player is range (but get auto played when player starts dialog) use the `Npc_IsInState(self,ZS_Talk)` trick. Lobart is also an example for an important message that gets played only when the player has started taking. That means AI_ProcessInfos must be called before the state ZS_TALK is set. It is currently called by doAIState: https://github.com/REGoth-project/REGoth/blob/ddeba8055eedafa39f45b13ff421b5d65651c4e1/src/logic/NpcScriptState.cpp#L253
```
FUNC INT DIA_Raoul_Hello_Condition()
{
	if (other.guild == GIL_NONE)
	&& (Npc_IsInState(self,ZS_Talk))
	{
		return TRUE;
	};
};
```
Regarding the wrong "What do you want this time?!" line, it seems like `self.aivar[AIV_TalkedToPlayer]` doesn't work.
```
FUNC VOID DIA_Raoul_Hello_Info()
{	
	if (self.aivar[AIV_TalkedToPlayer] == FALSE)
	{
		AI_Output (self ,other,"DIA_Raoul_Hello_01_00"); //(gelangweilt) Was willst du?
	}
	else
	{
		AI_Output (self ,other,"DIA_Raoul_Hello_01_01"); //(genervt) Was willst du schon wieder?
	};
};
```

**Comment by markusobi** [comment]
> so probably the wrong is picked.

The `aivar` issue is fixed now with #180 . He now uses his line "What do you want?" at first visit.
The `Npc_IsInState(self,ZS_Talk)` issue remains.
